synfig-studio/src/gui/dialogs/canvasresize.cpp

```diff
@@ -259,6 +262,27 @@ void CanvasResize::on_height_changed()
 	refresh_wh_toggle_widgets();
 }
 
+void CanvasResize::on_size_template_changed(){
+	String selection(combo_box->get_active_id());
+	if(selection == "custom") {
+		width->set_sensitive(true);
+		height->set_sensitive(true);
+		return;
+	}
+	String::size_type locx=selection.find_first_of('x');
+	String::size_type strSize=selection.size();
+	String x_size(selection.substr(0,locx));
+	String y_size(selection.substr(locx+1,strSize-1));
+	int x=stoi(x_size);
+	int y=stoi(y_size);
```

Comment:
`stoi()` [raises exception](https://en.cppreference.com/w/cpp/string/basic_string/stol) if string can't be converted. It is a controlled environment (internal ComboBox item id), but...

Reply:
should i use long instead?

Reply:
No, you don't need to. Well, the exception won't be thrown if the combobox item ids are well formed, so it is just  preciousness of mine. Let it be as it is.